app/js/docker/dockerService.js

```diff
@@ -66,6 +66,11 @@ function dockerService($http, actions, flux, alertService, configService) {
             }
         };
 
+        if (action === 'pullImage') {
+            // hack for pulling images
+            config.params = data;
+        }
+
```

Comment:
That's fair, I also assumed that all the post data would be in the body.
Either way sounds good to me.  I like how the first does the heavy lifting for me, but at the same time I like the customizability of the second.